state/unit_test.go

```diff
@@ -774,11 +779,46 @@ func (s *UnitSuite) TestShortCircuitDestroyUnit(c *gc.C) {
 	assertRemoved(c, s.unit)
 }
 
+func (s *UnitSuite) TestShortCircuitDestroyUnitNotAssigned(c *gc.C) {
```

Comment:
please add a test that concurrently assigns the unit to a machine when the unit is in error state